app/controllers/concerns/foreman/controller/authentication.rb

```diff
@@ -6,12 +6,13 @@ def available_sso
   end
 
   def authenticate
-    return true if (User.current && Rails.env.test? && api_request?) ||
-                   session[:user] && (User.current = User.unscoped.find_by(id: session[:user]))
+    User.current = User.unscoped.find_by(id: session[:user]) if session[:user]
+    return false if User.current&.disabled?
+    return true if (User.current && Rails.env.test? && api_request?) || User.current
```

Comment:
this postpone the `return` for ` Rails.env.test? && api_request?` after the assignment in comparison to previous code. Is it something we should be concerned about? May some test change user because of that?